src/jdk.jfr/share/classes/jdk/jfr/internal/MetadataRepository.java

```diff
@@ -218,7 +218,7 @@ private EventConfiguration makeConfiguration(Class<? extends jdk.internal.event.
         EventConfiguration configuration = newEventConfiguration(eventType, ec, settings);
         PlatformEventType pe = configuration.getPlatformEventType();
         pe.setRegistered(true);
-        if (jvm.isInstrumented(eventClass)) {
+        if (jvm.isInstrumented(eventClass) || !Utils.shouldInstrument(pe.isJDK(), pe.getName())) {
```

Comment:
This will call `setInstrumented()` even for events which are not really instrumented when they are not required to be instrumented, correct?

Reply:
Yes, I added comment to make it more clear.